src/main/java/htsjdk/samtools/SAMSequenceDictionary.java

```diff
@@ -266,5 +268,100 @@ public String toString() {
                 " length:"+ getReferenceLength()+" "+
                 " md5:"+md5()+")";
     }
+
+    public static final List<String> DEFAULT_DICTIONARY_EQUAL_TAG = Arrays.asList(
+            SAMSequenceRecord.MD5_TAG,
+            SAMSequenceRecord.SEQUENCE_LENGTH_TAG);
+
+    /**
+     * Will merge dictionaryTags from two dictionaries into one focusing on merging the tags rather than the sequences.
+     *
+     * Requires that dictionaries have the same SAMSequence records in the same order.
+     * For each sequenceIndex, the union of the tags from both sequences will be added to the new sequence, mismatching
+     * values (for tags that are in both) will generate a warning, and the value from dict1 will be used.
+     * For tags that are in tagsToEquate. an unequal value will generate an error (an IllegalArgumentException will
+     * be thrown.)
+     *
+     * @param dict1 first dictionary
+     * @param dict2 first dictionary
+     * @param tagsToMatch list of tags that must be equal if present in both sequence. Typical values will be MD, and LN
+     * @return dictionary consisting of the same sequences as the two inputs with the merged values of tags.
+     */
+    static public SAMSequenceDictionary mergeDictionaries(final SAMSequenceDictionary dict1,
+                                                          final SAMSequenceDictionary dict2,
+                                                          final List<String> tagsToMatch) {
```

Comment:
Circling back, I think this parameter should probably go away.  Length (if present) and MD5 (if present) should be _required_ to match in all cases, since by definition they should match.  Other tags, I'm not sure it matters if they match.